qemu/tests/live_backup_base.py

```diff
@@ -0,0 +1,210 @@
+                self.vm.monitor.transaction(args_list)
+                if not self.get_status():
+                    raise error.TestFail("%s backup job not found" % sync)
+                return None
+            error.context("Create bitmap for %s" % drive_name, logging.info)
+            self.vm.monitor.operate_dirty_bitmap("add", drive_name, bitmap_name, granularity)
+        error.context("Create %s backup for %s" % (sync, drive_name), logging.info)
+        if not backup_image_name:
+            raise error.TestError("No backup target provided.")
+        self.vm.monitor.drive_backup(drive_name, backup_image_name, backup_format,
+                                     sync, speed, mode, bitmap_name)
+        if not self.get_status():
+            raise error.TestFail("%s backup job not found" % sync)
+
+    def transaction_add(self, args_list, type, data):
+        """
+        Generate a args list for a transaction.
+        :param args_list: the arg_list param that will be modified
+        :param type: type of the command
+        :param data: data of the command
+        """
+        args = {"type": type, "data": data}
+        args_list.append(args)
+
+    def backup_check(self, compare_image=None):
+        """
+        Check and compare the backup images with qemu-img
+        :param compare_image: the image that need to be compared
+        """
+        data_dir = self.data_dir
+        for image in self.image_chain:
+            params = self.params.object_params(image)
+            qemu_image = qemu_storage.QemuImg(params, data_dir, image)
+            qemu_image.check_image(params, data_dir)
+        params = self.params.object_params(self.source_image)
+        if compare_image:
+            compare_image = utils_misc.get_path(data_dir, compare_image)
+            source_image = qemu_storage.storage.get_image_filename(params, data_dir)
+            qemu_image.compare_images(compare_image, source_image)
+
+    def reopen(self):
+        """
+        Closing the vm and reboot it with the backup image.
+        """
+        vm = self.vm
+        vm.destroy()
+        image_chain = self.image_chain
+        image_name = self.params.get("image_name_%s" %
+                                     image_chain[len(image_chain) - 1])
+        self.params["image_name_%s" % self.source_image] = image_name
+        vm.create(params=self.params)
+        self.vm = vm
+        session = self.get_session()
+        session.close()
+
+    def before_full_backup(self):
+        """
+        Run steps before create full backup.
+        """
+        return self.do_steps("before_full_backup")
+
+    def before_incremental(self):
+        """
+        Run steps before create incremental backup.
+        """
+        return self.do_steps("before_incremental")
+
+    def after_full_backup(self):
+        """
+        Run steps after create full backup.
+        """
+        self.do_steps("after_full_backup")
+        time.sleep(120)
```

Comment:
@xutian All test cases of this feature have steps like "full backup" -> "incremental backup", I need guest generate dirty page between "full backup" and "incremental backup", so that the "incremental backup" will record something, other wise 'incremental backup' will be idle.